rosette/src/Export.cc

```diff
@@ -575,17 +621,166 @@ void rblTableObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::O
     populateObjectByType(rt->tbl, tbl);
 }
 
+void ctxtObjectHandler( ObjectCodePB::Object * pbob, pOb ob, ObjectCodePB::ObType pbtype) {
+    ObjectCodePB::Ctxt *ctxtob = pbob->mutable_ctxt();
+    Ctxt * c = (Ctxt *)ob;
+
+    ObjectCodePB::Location *tag = ctxtob->mutable_tag();
+    populateLocation(c->tag, tag);
+
+    ctxtob->set_nargs( c->nargs );
+    ctxtob->set_outstanding( c->outstanding );
+    ctxtob->set_pc( c->pc );
+
+    ObjectCodePB::Object *rslt = ctxtob->mutable_rslt();
+    populateObjectByType(c->rslt, rslt);
+
+    ObjectCodePB::Object *trgt = ctxtob->mutable_trgt();
+    populateObjectByType(c->trgt, trgt);
+
+    ObjectCodePB::Object *argvec = ctxtob->mutable_argvec();
+    populateObjectByType(c->argvec, argvec);
+
+    ObjectCodePB::Object *env = ctxtob->mutable_env();
+    populateObjectByType(c->env, env);
+
+    ObjectCodePB::Object *code = ctxtob->mutable_code();
+    populateObjectByType(c->code, code);
+
+    ObjectCodePB::Object *ctxt = ctxtob->mutable_ctxt();
+    populateObjectByType(c->ctxt, ctxt);
+
+    ObjectCodePB::Object *self2 = ctxtob->mutable_self2();
+    populateObjectByType(c->self2, self2);
+
+    ObjectCodePB::Object *selfenv = ctxtob->mutable_selfenv();
+    populateObjectByType(c->selfEnv, selfenv);
+
+    ObjectCodePB::Object *rcvr = ctxtob->mutable_rcvr();
+    populateObjectByType(c->rcvr, rcvr);
+
+    // ObjectCodePB::Object *monitor = ctxtob->mutable_monitor();
+    // populateObjectByType(c->monitor, monitor);
+
+}
+
+void indexedMetaObjectHandler( ObjectCodePB::Object * pbob, pOb ob, ObjectCodePB::ObType pbtype) {
+    ObjectCodePB::IndexedMeta *imob = pbob->mutable_indexed_meta();
+    IndexedMeta * im = (IndexedMeta *)ob;
+
+    ObjectCodePB::Object *extension = imob->mutable_extension();
+    populateObjectByType(im->extension, extension);
+}
+
+void stdOprnObjectHandler( ObjectCodePB::Object * pbob, pOb ob, ObjectCodePB::ObType pbtype ) {
+    ObjectCodePB::StdOprn *soob = pbob->mutable_std_oprn();
+    StdOprn * so = (StdOprn *)ob;
+
+    ObjectCodePB::Object *extension = soob->mutable_extension();
```

Comment:
Looks good but as a side note: I haven't seen `StdOprn` making use of its `Actor` capabilities anywhere. Only saw it being used as a key so far. If you know about places where it is not just used as a key, I would be interested in that.

Reply:
I don't know of any specific uses. I just know when I started processing meta and parent pointers, this showed up as being needed. Whenever I go to process a pOb, if its type isn't implemented, I log a warning. I then go and add support for it and in some cases I've added support for the Objects from which it is derived.  In my next go around, I plan to better and more fully support object inheritance.